### PR TITLE
Deprecate imports from traitsui.editors

### LINKED ISSUE #1617: Cleanup traitsui.editors init module
The commit https://github.com/enthought/traitsui/commit/1d3962e97b3b85c1ac560d7473589304b461b94a seems to have re-exposed `traitsui.editors.api` imports via the `traitsui.editors.__init__` module. This seems to have been done for backwards compatibility - and i don't think this is needed anymore.

https://github.com/enthought/traitsui/blob/6913fa460be6d2a6bf8dbd1320144c71a97014b2/traitsui/editors/__init__.py

### PR DESCRIPTION
**Pull request by corranwebster**
Imports should be from traitsui.editors.api.  This adds a simple test to ensure imports expected.

Fixes #1617

**Checklist**
- [ ] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

**Comment by corranwebster** [comment]
Converting to draft since this requires Python 3.7 - depends on #1994

File: traitsui/tests/test_editors_imports.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Thanks for using Enthought open source!
+
+import unittest
+
+
+class TestEditorsImports(unittest.TestCase):
+
+    def test_editors_import_warns(self):
+        # Importing from pyface.editors is deprecated
```
Comment (dpinte) [inline traitsui/tests/test_editors_imports.py:17]: traitsui, not pyface?

**Review by dpinte — APPROVED** [review]
@corranwebster one minor question. Otherwise, looks good to me.

File: traitsui/tests/test_editors_imports.py (hunk 1 above)
Comment (corranwebster) [inline traitsui/tests/test_editors_imports.py:17]:
```suggestion
        # Importing from traitsui.editors is deprecated
```

**Pull request merged**